src/sqlancer/ProviderAdapter.java

```diff
@@ -41,11 +43,14 @@ public void generateAndTestDatabase(G globalState) throws SQLException {
 
             TestOracle oracle = getTestOracle(globalState);
             for (int i = 0; i < globalState.getOptions().getNrQueries(); i++) {
-                try {
-                    oracle.check();
-                    globalState.getManager().incrementSelectQueryCount();
-                } catch (IgnoreMeException e) {
+                try (OracleRunReproductionState localState = globalState.getState().createLocalState()) {
```

Comment:
Is the` localState.close() ` function automatically called after the `try` block is executed here? Or does the `localState.close()` function need to be explicitly called after the block is executed? I am not exactly familiar with how the `Closeable `interface functions in that regard.

Reply:
By using the `try` block, the `close()` method is automatically called when execution leaves this block. I'm not so happy about the magic that is currently involved in this (it's quite in transparent how and when what happens), but I think it's an improvement in the sense that it generalizes the previous `queryString` by allowing to log multiple statements and not having to manually reset the state.